contracts/utils/math/Math.sol

```diff
@@ -223,4 +195,146 @@ library Math {
         }
         return result;
     }
+
+    /**
+     * @dev Return the log in base 2, rounded down, of a positive value.
+     * Returns 0 if given 0.
+     */
+    function log2(uint256 value) internal pure returns (uint256) {
+        uint256 result = 0;
+        if (value >> 128 > 0) {
+            value >>= 128;
+            result += 128;
+        }
+        if (value >> 64 > 0) {
+            value >>= 64;
+            result += 64;
+        }
+        if (value >> 32 > 0) {
+            value >>= 32;
+            result += 32;
+        }
+        if (value >> 16 > 0) {
+            value >>= 16;
+            result += 16;
+        }
+        if (value >> 8 > 0) {
+            value >>= 8;
+            result += 8;
+        }
+        if (value >> 4 > 0) {
+            value >>= 4;
+            result += 4;
+        }
+        if (value >> 2 > 0) {
+            value >>= 2;
+            result += 2;
+        }
+        if (value >> 1 > 0) {
+            result += 1;
+        }
+        return result;
+    }
+
+    /**
+     * @dev Return the log in base 2, following the selected rounding direction, of a positive value.
+     * Returns 0 if given 0.
+     */
+    function log2(uint256 value, Rounding rounding) internal pure returns (uint256) {
+        uint256 result = log2(value);
+        if (rounding == Rounding.Up && 1 << result < value) {
+            result += 1;
+        }
+        return result;
+    }
+
+    /**
+     * @dev Return the log in base 10, rounded down, of a positive value.
+     * Returns 0 if given 0.
+     */
+    function log10(uint256 value) internal pure returns (uint256) {
+        uint256 result = 0;
+        if (value >= 10**64) {
```

Comment:
@Amxx you can replace `>=` by `>` and save more gas.

Example: 
```
value >= 10**64 == value > 10**64-1
```

Reply:
@Amxx what about this one?

Reply:
I tested both, and your version is more expensive :/ it looks like the compiler does the `-1` operation at runtime